src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.Android.cs

```diff
@@ -29,11 +36,42 @@ internal static void SwapViews(UIElement oldView, UIElement newView)
 					{
 						newContentAsFrameworkElement.TemplatedParent = (unoViewGroup as IFrameworkElement)?.TemplatedParent;
 					}
-					unoViewGroup.AddView(newView, currentPosition.Value);
+					InsertViewAt(unoViewGroup, currentPosition.Value, newView);
```

Comment:
The calls to InsertViewAt can be moved out of this block and made into a single call since unoViewGroup and parentViewGroup are the same object and InsertViewAt just accepts a ViewGroup